### ISSUE TITLE
Can we have ObserverManager implemented in Orleans

### ISSUE THREAD

**Issue by j0nimost** [dotnet/orleans#7858]
While following the [Docs](https://docs.microsoft.com/en-us/dotnet/orleans/grains/observers)  on how to implement [Observers](https://docs.microsoft.com/en-us/dotnet/orleans/grains/observers#code-example); 
I noticed that `ObserverManager` is not defined in the Orleans package but instead its defined here:
[ObserverManager](https://github.com/dotnet/orleans/blob/e997335d2d689bb39e67f6bcf6fd70862a22c02f/test/Grains/TestGrains/ObserverManager.cs)

**Comment by JorgeCandeias** [comment]
TLDR; there's [an open item to resurrect that long lost friend for v4.x](https://github.com/dotnet/orleans/issues/7099) so there's a possibility you may have your wish granted.

Long story, the `ObserverSubscriptionManager` was moved to a legacy package back in v2.x, you can see the code here:
https://github.com/dotnet/orleans/blob/v2.4.5/src/Orleans.Core.Legacy/Async/ObserverSubscriptionManager.cs

It doesn't look like that code survived the cut to v3.x, let alone to v4.x, and that's probably for the best. That implementation will root subscriptions that die off without unsubscribing, so you end up with memory leaks over time if there's nothing to notify. There's a best effort to remove them on failure to reach them but that only works if there's something to send to them. And then if those subs are just experiencing a wobbly and are not dead, the "connection" breaks without recovery. It's not a reliable cycle. So until the team or a contributor writes a more robust implementation, it's best to write your own custom manager, with active cleanup logic specific to your use case. Or better yet, just use streams, the memory provider works like a charm for these things if you don't care about persisting messages.

**Comment by j0nimost** [comment]
I really wanted to implement a subscription model since my instance I would like to trigger it. It's so unfortunate that they left it since in this particular case I want to use it within Orleans and not download some CQRS package. The documentation is SO BAD and scanty that you have to literary stumble upon a feature. I have no clue where I start in writing an implementation of an `ObserverManager`.

**Comment by ReubenBond** [comment]
We should also bring ObserverManager back. There is a version from the tests which you ought to be able to copy into your project: https://github.com/dotnet/orleans/blob/main/test/Grains/TestGrains/ObserverManager.cs

**Comment by ReubenBond** [comment]
A PR for this would be welcome, by the way, to move ObserverManager into Orleans.Core
